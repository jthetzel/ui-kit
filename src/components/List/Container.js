import styled from 'react-emotion';

export default styled.ul`
  display: block;
  width: 200px;
  background: white;
  padding: 10px 0;
  
  font-family: 'Noto Sans', serif;
  font-size: 16px;

  margin: 0;
`;
