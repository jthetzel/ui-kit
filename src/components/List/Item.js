import styled from 'react-emotion';

const Item = styled.li`
  position: relative;
  display: flex;
  list-style-type: none;
  height: 50px;
  padding: 10px;
  box-sizing: border-box;
  align-items: center;
  user-select: none;
  background: ${props => (props.isActive ? '#f5f4f5' : '#FFFFFF')};

  &:after {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    height: 1px;
    margin: 0 10px;
    background: ${props => (props.isActive ? '#f5f4f5' : '#faf9fa')};
    content: ' ';
  }

  &:hover {
    color: #223bf2;
    background: #faf9fa;
    cursor: pointer;
  }

  &:last-child {
    &:after {
      background: transparent;
    }
  }
`;

export default Item;
