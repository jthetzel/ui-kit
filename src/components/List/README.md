Simple Items List with highlight support

```js
const items = [
  { value: "one", label: "One", highlight: true },
  { value: "two", label: "Two" },
  { value: "three", label: "Three", highlight: true },
  { value: "four", label: "Four" }
];

<List items={items} />;
```
