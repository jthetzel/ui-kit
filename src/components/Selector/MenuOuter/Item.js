import styled from 'react-emotion';

export default styled.li`
  display: flex;
  list-style-type: none;
  height: 50px;
  border-bottom: 1px solid gray;
  padding: 10px;
  box-sizing: border-box;
  align-items: center;
  user-select: none;
  
  background-color: ${({ isActive }) => (isActive ? '#f5f4f5' : '#FFFFFF')};

  &:hover {
    color: #223bf2;
  }

  &:last-child {
    border-bottom: 0;
  }
`;
