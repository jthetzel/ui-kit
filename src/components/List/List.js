import React from 'react';
import PropTypes from 'prop-types';

import Container from './Container';
import Item from './Item';

const List = ({ items, onClick }) => (
  <Container>
    {items.map(item => (
      <Item
        key={item.value}
        highlight={item.highlight}
        onClick={onClick && onClick.bind(null, item)}
      >
        {item.label}
      </Item>
    ))}
  </Container>
);

List.propTypes = {
  onClick: PropTypes.func,
  items: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    label: PropTypes.string.isRequired,
    highlight: PropTypes.bool
  })).isRequired
};

List.defaultProps = {
  onClick: null
};

export default List;
