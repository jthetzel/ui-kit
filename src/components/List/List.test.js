import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import List from '../List';
import Item from './Item';

const items = [
  { value: 'one', label: 'One' },
  { value: 'two', label: 'Two' },
  { value: 'three', label: 'Three' },
  { value: 'four', label: 'Four' }
];

const mockCallback = jest.fn();
const onClick = value => mockCallback(value);

test('List does not crash', () => {
  const tree = shallow(<List onClick={() => {}} items={items} value={items[0].value} />);
  expect(toJson(tree)).toMatchSnapshot();
});

test('List simulate onClickItem', () => {
  const wrapper = shallow(<List onClick={onClick} items={items} value={items[0].value} />);
  const scaleButton = wrapper.find(Item).last();
  scaleButton.simulate('click');
  expect(mockCallback.mock.calls[0][0].value).toBe('four');
});

test('List change value', () => {
  const wrapper = mount(<List onClick={onClick} items={items} value={items[0].value} />);
  expect(wrapper.props().value).toBe('one');
  wrapper.setProps({ value: items[3].value });
  expect(wrapper.props().value).toBe('four');
});
