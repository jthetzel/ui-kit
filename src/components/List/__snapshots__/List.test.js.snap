// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List does not crash 1`] = `
<Styled(ul)>
  <Styled(li)
    highlight={false}
    key="one"
    onClick={[Function]}
  >
    One
  </Styled(li)>
  <Styled(li)
    highlight={false}
    key="two"
    onClick={[Function]}
  >
    Two
  </Styled(li)>
  <Styled(li)
    highlight={false}
    key="three"
    onClick={[Function]}
  >
    Three
  </Styled(li)>
  <Styled(li)
    highlight={false}
    key="four"
    onClick={[Function]}
  >
    Four
  </Styled(li)>
</Styled(ul)>
`;
